airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -142,6 +144,33 @@ def _validate_source(self):
         except ValidationError as e:
             raise ValidationError("Validation against json schema defined in declarative_component_schema.yaml schema failed") from e
 
+        cdk_version = metadata.version("airbyte_cdk")
+        cdk_major, cdk_minor, cdk_patch = (int(part) for part in cdk_version.split("."))
+        manifest_version = self._source_config.get("version")
+        manifest_major, manifest_minor, manifest_patch = self._get_version_parts(manifest_version)
+
+        if cdk_major < manifest_major or (cdk_major == manifest_major and cdk_minor < manifest_minor):
+            raise ValidationError(
+                f"The manifest version {manifest_version} is greater than the airbyte-cdk package version ({cdk_version}). Your "
+                f"manifest may contain features that are not in the current CDK version."
+            )
+        elif manifest_major == 0 and manifest_minor < 29 and (0 < cdk_major or 29 <= cdk_minor):
```

Comment:
the `(0 < cdk_major or 29 <= cdk_minor)` part of this condition is unnecessary because it can never get false (we will never go back before 0.29 from this point on)

Reply:
I am not sure that is completely true. We _could_ have an individual connector that pins it's CDK version back to `0.25.0` and the manifest is also running below `0.29.0`. In that case we would not want to throw an error because it is technically a valid pre-beta manifest. Basically it covers the case that manifest + cdk are pre-beta

Reply:
But this change won’t be part of cdk version 0.25.0 - it will go in 0.31.0 or so and onward.

Do you plan to release 0.25.1 separately which cherry-picks this check back to the state of 0.25.0?

Reply:
ah duh yeah you're right the old cdk version wouldn't have this logic. i'll get rid of it